packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -372,6 +370,137 @@ export class CloudFormationDeployments {
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, deployedTemplate: any, sdk: ISDK,
+  ): Promise<void> {
+    await this.replaceNestedStacksInParentTemplate(
+      rootStackArtifact, rootStackArtifact.template, deployedTemplate, rootStackArtifact.stackName, sdk,
+    );
+  }
+
+  private async replaceNestedStacksInParentTemplate(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact,
+    generatedParentTemplate: any,
+    deployedParentTemplate: any,
+    parentStackName: string | undefined,
+    sdk: ISDK,
+  ): Promise<void> {
+    const listStackResources = parentStackName ? new LazyListStackResources(sdk, parentStackName) : undefined;
+    const parentNestedStackResources = Object.entries(generatedParentTemplate.Resources ?? {});
+    for (const [nestedStackLogicalId, resource] of parentNestedStackResources) {
+      if (!this.isCdkManagedNestedStack(resource)) {
+        continue;
+      }
+
+      const nestedStackResource: NestedStackResource = resource;
+      const assetPath = nestedStackResource.Metadata['aws:asset:path'];
+      const nestedStackTemplates = await this.getNestedStackTemplates(
+        rootStackArtifact, assetPath, nestedStackLogicalId, parentStackName, listStackResources, sdk,
+      );
+
+      const populatedNestedTemplates = this.populateParentTemplates(
+        generatedParentTemplate.Resources[nestedStackLogicalId],
+        (deployedParentTemplate.Resources ?? {})[nestedStackLogicalId],
+        nestedStackTemplates,
+      );
+
+      await this.replaceNestedStacksInParentTemplate(
+        rootStackArtifact,
+        populatedNestedTemplates.generatedNestedTemplate,
+        populatedNestedTemplates.deployedNestedTemplate,
+        nestedStackTemplates.nestedStackName,
+        sdk,
+      );
+    }
+  }
+
+  private async getNestedStackTemplates(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, nestedTemplateAssetPath: string, nestedStackLogicalId: string,
+    parentStackName: string | undefined, listStackResources: ListStackResources | undefined, sdk: ISDK,
+  ): Promise<NestedStackTemplates> {
+    const nestedTemplatePath = path.join(rootStackArtifact.assembly.directory, nestedTemplateAssetPath);
+    // CFN generates the nested stack name in the form `ParentStackName-NestedStackLogicalID-SomeHashWeCan'tCompute,
+    // so we get the ARN and manually extract the name.
+    //const nestedStackArn = (await listStackResources?.listStackResources())?.find(sr => sr.LogicalResourceId === nestedStackLogicalId)?.PhysicalResourceId;
+    let nestedStackArn: string | undefined = undefined;
+
+    try {
+      const stackResources = await listStackResources?.listStackResources();
+
+      if (stackResources) {
+        nestedStackArn = stackResources.find(sr => sr.LogicalResourceId === nestedStackLogicalId)?.PhysicalResourceId;
+      }
+    } catch (e) {
+      if (e.message !== `Stack with id ${parentStackName} does not exist`) {
+        throw e;
+      }
+    }
```

Comment:
I think you probably need a helper method here, `getNestedStackArn(nestedStackLogicalId: string, listStackResource: ListStackResources | undefined): string | undefined`, or something like that.